vllm_ascend/ops/rotary_embedding.py

```diff
@@ -87,5 +87,24 @@ def rope_deepseek_forward_oot(
     return query, key
 
 
+def mrope_forward(
+    self,
+    positions: torch.Tensor,
+    query: torch.Tensor,
+    key: torch.Tensor,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    import torch_npu
+    mrope_section = [0, 0, 0] if positions.ndim == 1 else self.mrope_section
+    query, key = torch_npu.npu_mrope(positions,
+                                     query.contiguous(),
+                                     key.contiguous(),
+                                     self.cos_sin_cache.contiguous(),
+                                     self.head_size,
+                                     mrope_section=mrope_section,
+                                     rotary_mode='half')
+    return query, key
+
+
 RotaryEmbedding.forward_oot = rope_forward_oot
+MRotaryEmbedding.forward = mrope_forward
```

Comment:
basically, we don't want to change anything in vllm except `foward_oot`. For this kind of change, we should ask vllm to satisfy our requirement. Let's do it in the future. Thanks.